src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -150,8 +150,15 @@ private void ProcessOptions(IProjectRuleSnapshot snapshot)
         {
             Assumes.NotNull(_context);
 
+            // The order of options matters and we should use warnaserror- and then warnaserror+
+            // Depending which comes first, the compiler produces different diagnostics.
+            // We just need to move warnaserror+ to the end.
+            var warnaserrorPlus = snapshot.Items.Where(a => a.Key.Contains("/warnaserror+")).FirstOrDefault();
```

Comment:
nit: `FirstOrDefault` allows a `Where` condition in it. So, this can just be:
```suggestion
            var warnaserrorPlus = snapshot.Items.FirstOrDefault(a => a.Key.Contains("/warnaserror+"));
```